web/src/lib/components/shared-components/profile-image-cropper.svelte

```diff
@@ -0,0 +1,85 @@
+<script lang="ts">
+  import { AssetResponseDto, api } from '@api';
+  import { createEventDispatcher } from 'svelte';
+  import { notificationController, NotificationType } from './notification/notification';
+  import { handleError } from '$lib/utils/handle-error';
+  import domtoimage from 'dom-to-image';
+  import PhotoViewer from '../asset-viewer/photo-viewer.svelte';
+  import BaseModal from './base-modal.svelte';
+  import Button from '../elements/buttons/button.svelte';
+
+  export let asset: AssetResponseDto;
+  export let publicSharedKey = '';
+
+  const dispatch = createEventDispatcher();
+  let imgElement: HTMLDivElement;
+
+  const hasTransparentPixels = async (blob: Blob) => {
+    const img = new Image();
+    img.src = URL.createObjectURL(blob);
+    return new Promise((resolve) => {
+      img.onload = function () {
+        const canvas = document.createElement('canvas');
+        canvas.width = img.width;
+        canvas.height = img.height;
+        const ctx = canvas.getContext('2d');
+        ctx?.drawImage(img, 0, 0);
+        const imageData = ctx?.getImageData(0, 0, canvas.width, canvas.height);
+        const data = imageData?.data;
+        if (!data) {
+          throw new Error('Could not get image data.');
+        }
+        for (let i = 0; i < data.length; i += 4) {
+          if (data[i + 3] < 255) {
+            resolve(true);
+            return;
+          }
+        }
+        resolve(false);
+      };
+    });
+  };
+
+  const handleSetProfilePicture = async () => {
+    try {
+      const blob = await domtoimage.toBlob(imgElement);
+      if (await hasTransparentPixels(blob)) {
+        notificationController.show({
+          type: NotificationType.Error,
+          message: 'Profile pictures cannot have transparent pixels.',
```

Comment:
why?

Reply:
I think transparent areas, caused by bad positioning, are not really pleasing in a profile picture. As i havent found an easy way to restrict the profile picture selection the the image area itself, thats what i came up with for now. But it might not be necessary.